# College Majors Project
### By Charlotte Burbridge

In [1]:
import pandas as pd
from pandas_profiling import ProfileReport

In [2]:
majorsdf_raw = pd.read_csv('majors.csv')
majorsdf_raw

Rank  Major_code                                      Major    Total  \
0       1        2419                      PETROLEUM ENGINEERING   2339.0   
1       2        2416             MINING AND MINERAL ENGINEERING    756.0   
2       3        2415                  METALLURGICAL ENGINEERING    856.0   
3       4        2417  NAVAL ARCHITECTURE AND MARINE ENGINEERING   1258.0   
4       5        2405                       CHEMICAL ENGINEERING  32260.0   
..    ...         ...                                        ...      ...   
168   169        3609                                    ZOOLOGY   8409.0   
169   170        5201                     EDUCATIONAL PSYCHOLOGY   2854.0   
170   171        5202                        CLINICAL PSYCHOLOGY   2838.0   
171   172        5203                      COUNSELING PSYCHOLOGY   4626.0   
172   173        3501                            LIBRARY SCIENCE   1098.0   

         Men    Women            Major_category  ShareWomen  Sample_size  \
0     2057.0    282.0               Engineering    0.120564           36   
1      679.0     77.0               Engineering    0.101852            7   
2      725.0    131.0               Engineering    0.153037            3   
3     1123.0    135.0               Engineering    0.107313           16   
4    21239.0  11021.0               Engineering    0.341631          289   
..       ...      ...                       ...         ...          ...   
168   3050.0   5359.0    Biology & Life Science    0.637293           47   
169    522.0   2332.0  Psychology & Social Work    0.817099            7   
170    568.0   2270.0  Psychology & Social Work    0.799859           13   
171    931.0   3695.0  Psychology & Social Work    0.798746           21   
172    134.0    964.0                 Education    0.877960            2   

     Employed  ...  Part_time  Full_time_year_round  Unemployed  \
0        1976  ...        270                  1207          37   
1         640  ...        170                   388          85   
2         648  ...        133                   340          16   
3         758  ...        150                   692          40   
4       25694  ...       5180                 16697        1672   
..        ...  ...        ...                   ...         ...   
168      6259  ...       2190                  3602         304   
169      2125  ...        572                  1211         148   
170      2101  ...        648                  1293         368   
171      3777  ...        965                  2738         214   
172       742  ...        237                   410          87   

     Unemployment_rate  Median  P25th   P75th  College_jobs  Non_college_jobs  \
0             0.018381  110000  95000  125000          1534               364   
1             0.117241   75000  55000   90000           350               257   
2             0.024096   73000  50000  105000           456               176   
3             0.050125   70000  43000   80000           529               102   
4             0.061098   65000  50000   75000         18314              4440   
..                 ...     ...    ...     ...           ...               ...   
168           0.046320   26000  20000   39000          2771              2947   
169           0.065112   25000  24000   34000          1488               615   
170           0.149048   25000  25000   40000           986               870   
171           0.053621   23400  19200   26000          2403              1245   
172           0.104946   22000  20000   22000           288               338   

     Low_wage_jobs  
0              193  
1               50  
2                0  
3                0  
4              972  
..             ...  
168            743  
169             82  
170            622  
171            308  
172            192  

[173 rows x 21 columns]

In [8]:
ProfileReport(majorsdf_raw).to_file("majorsPR.html")

Summarize dataset:   0%|          | 0/35 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

## questions
- which one has highest employment? which has highest median salary? are these the same or part of the same major category?
- majors with high percentage of women in relation to salary, is there a correlation?
- percentage of women in engineering vs stem majors in total, which stem major has the highest amount of women

## How many people were interviewed for this data?

In [15]:
majorsdf_raw.Total.sum()

6771654.0

### Here we can see that 6,771,654 people were interviewed for this data.





## What major has the lowest unemployment rate? And which has the highest?

In [3]:
min_emp = majorsdf_raw['Unemployment_rate'].min()
majorsdf_raw.loc[majorsdf_raw.Unemployment_rate == min_emp]

Rank  Major_code                                       Major   Total  \
52     53        4005            MATHEMATICS AND COMPUTER SCIENCE   609.0   
73     74        3801                       MILITARY TECHNOLOGIES   124.0   
83     84        3602                                      BOTANY  1329.0   
112   113        1106                                SOIL SCIENCE   685.0   
120   121        2301  EDUCATIONAL ADMINISTRATION AND SUPERVISION   804.0   

       Men  Women                       Major_category  ShareWomen  \
52   500.0  109.0              Computers & Mathematics    0.178982   
73   124.0    0.0  Industrial Arts & Consumer Services    0.000000   
83   626.0  703.0               Biology & Life Science    0.528969   
112  476.0  209.0      Agriculture & Natural Resources    0.305109   
120  280.0  524.0                            Education    0.651741   

     Sample_size  Employed  ...  Part_time  Full_time_year_round  Unemployed  \
52             7       559  ...          0                   391           0   
73             4         0  ...          0                   111           0   
83             9      1010  ...        169                   740           0   
112            4       613  ...        185                   383           0   
120            5       703  ...          0                   504           0   

     Unemployment_rate  Median  P25th  P75th  College_jobs  Non_college_jobs  \
52                 0.0   42000  30000  78000           452                67   
73                 0.0   40000  40000  40000             0                 0   
83                 0.0   37000  26000  40000           677               184   
112                0.0   35000  18500  44000           355               144   
120                0.0   34000  29000  35000           346               206   

     Low_wage_jobs  
52              25  
73               0  
83              56  
112              0  
120            111  

[5 rows x 21 columns]

### The majors with the lowest unemployment rate are Mathmatics and Computer Science, Military Technologies, Botany, Soil Science, and Educational Administration and Supervision

## Which majors have the highest unemployment rate?

In [4]:
max_emp = majorsdf_raw['Unemployment_rate'].max()
majorsdf_raw.loc[majorsdf_raw.Unemployment_rate == max_emp]

Rank  Major_code                Major   Total     Men  Women  \
5     6        2418  NUCLEAR ENGINEERING  2573.0  2200.0  373.0   

  Major_category  ShareWomen  Sample_size  Employed  ...  Part_time  \
5    Engineering    0.144967           17      1857  ...        264   

   Full_time_year_round  Unemployed  Unemployment_rate  Median  P25th   P75th  \
5                  1449         400           0.177226   65000  50000  102000   

   College_jobs  Non_college_jobs  Low_wage_jobs  
5          1142               657            244  

[1 rows x 21 columns]

### The major with the highest Employment rate (about 18%) is Nuclear Engineering

## What about highest and lowest median salaries of these majors?

In [30]:
min_sal = majorsdf_raw['Median'].min()
majorsdf_raw.loc[majorsdf_raw.Median == min_sal]

Rank  Major_code            Major   Total    Men  Women Major_category  \
172   173        3501  LIBRARY SCIENCE  1098.0  134.0  964.0      Education   

     ShareWomen  Sample_size  Employed  ...  Part_time  Full_time_year_round  \
172     0.87796            2       742  ...        237                   410   

     Unemployed  Unemployment_rate  Median  P25th  P75th  College_jobs  \
172          87           0.104946   22000  20000  22000           288   

     Non_college_jobs  Low_wage_jobs  
172               338            192  

[1 rows x 21 columns]

### The Lowest Median Salary major is Library Science, with the median salary of 22,000 dollars

## And what is the majors with the highest median salary?

In [6]:
max_sal = majorsdf_raw['Median'].max()
majorsdf_raw.loc[majorsdf_raw.Median == max_sal]

Rank  Major_code                  Major   Total     Men  Women  \
0     1        2419  PETROLEUM ENGINEERING  2339.0  2057.0  282.0   

  Major_category  ShareWomen  Sample_size  Employed  ...  Part_time  \
0    Engineering    0.120564           36      1976  ...        270   

   Full_time_year_round  Unemployed  Unemployment_rate  Median  P25th   P75th  \
0                  1207          37           0.018381  110000  95000  125000   

   College_jobs  Non_college_jobs  Low_wage_jobs  
0          1534               364            193  

[1 rows x 21 columns]

### The Highest Median Salary is Petroleum Engineering, with the median salary of 110,000 dollars

## Which major category overall has highest percentage of Noncollege jobs out of the jobs reported?

### First, what are the categories?

In [17]:
majorsdf_raw.Major_category.unique()

array(['Engineering', 'Business', 'Physical Sciences',
       'Law & Public Policy', 'Computers & Mathematics',
       'Agriculture & Natural Resources',
       'Industrial Arts & Consumer Services', 'Arts', 'Health',
       'Social Science', 'Biology & Life Science', 'Education',
       'Humanities & Liberal Arts', 'Psychology & Social Work',
       'Communications & Journalism', 'Interdisciplinary'], dtype=object)

### How many majors are in each category?

In [26]:
majorsdf_raw['Major_category'].value_counts()

# geeksforgeeks helped me remember how to use value_counts
# https://www.geeksforgeeks.org/how-to-extract-the-value-names-and-counts-from-value_counts-in-pandas/
# forever grateful to geeksforgeeks

Engineering                            29
Education                              16
Humanities & Liberal Arts              15
Biology & Life Science                 14
Business                               13
Health                                 12
Computers & Mathematics                11
Physical Sciences                      10
Agriculture & Natural Resources        10
Social Science                          9
Psychology & Social Work                9
Arts                                    8
Industrial Arts & Consumer Services     7
Law & Public Policy                     5
Communications & Journalism             4
Interdisciplinary                       1
Name: Major_category, dtype: int64

### Now I'm making a new dataframe to make this easier for me to understand.

### This dataframe doesn't have the individual majors but it has the major categories. It also has the sum of all the jobs related to their degree and jobs not related to their degree per major category.

In [49]:
majorsdf2 = majorsdf_raw.groupby('Major_category')[['Non_college_jobs', 'College_jobs']].sum()
majorsdf2

Non_college_jobs  College_jobs
Major_category                                                     
Agriculture & Natural Resources                 34491         19860
Arts                                           163720         94785
Biology & Life Science                         127182        151233
Business                                       496570        148538
Communications & Journalism                    172992         86556
Computers & Mathematics                         74463        137859
Education                                      121761        338713
Engineering                                    102383        269767
Health                                         110496        245441
Humanities & Liberal Arts                      276532        192650
Industrial Arts & Consumer Services            104620         57200
Interdisciplinary                                3903          5176
Law & Public Policy                            100024         29221
Physical Sciences                               49469         76552
Psychology & Social Work                       169366        164305
Social Science                                 190246        113960

### Now I'm going to make a column with the percentage of noncollege jobs per major categories to be in the dataframe.

In [50]:
majorsdf2['Non_percentage'] = majorsdf2['Non_college_jobs'] / (majorsdf2['Non_college_jobs'] + majorsdf2['College_jobs'])
majorsdf2

Non_college_jobs  College_jobs  \
Major_category                                                        
Agriculture & Natural Resources                 34491         19860   
Arts                                           163720         94785   
Biology & Life Science                         127182        151233   
Business                                       496570        148538   
Communications & Journalism                    172992         86556   
Computers & Mathematics                         74463        137859   
Education                                      121761        338713   
Engineering                                    102383        269767   
Health                                         110496        245441   
Humanities & Liberal Arts                      276532        192650   
Industrial Arts & Consumer Services            104620         57200   
Interdisciplinary                                3903          5176   
Law & Public Policy                            100024         29221   
Physical Sciences                               49469         76552   
Psychology & Social Work                       169366        164305   
Social Science                                 190246        113960   

                                     Non_percentage  
Major_category                                       
Agriculture & Natural Resources            0.634597  
Arts                                       0.633334  
Biology & Life Science                     0.456807  
Business                                   0.769747  
Communications & Journalism                0.666513  
Computers & Mathematics                    0.350708  
Education                                  0.264425  
Engineering                                0.275112  
Health                                     0.310437  
Humanities & Liberal Arts                  0.589392  
Industrial Arts & Consumer Services        0.646521  
Interdisciplinary                          0.429893  
Law & Public Policy                        0.773910  
Physical Sciences                          0.392546  
Psychology & Social Work                   0.507584  
Social Science                             0.625385

In [51]:
majorsdf2.sort_values('Non_percentage')

Non_college_jobs  College_jobs  \
Major_category                                                        
Education                                      121761        338713   
Engineering                                    102383        269767   
Health                                         110496        245441   
Computers & Mathematics                         74463        137859   
Physical Sciences                               49469         76552   
Interdisciplinary                                3903          5176   
Biology & Life Science                         127182        151233   
Psychology & Social Work                       169366        164305   
Humanities & Liberal Arts                      276532        192650   
Social Science                                 190246        113960   
Arts                                           163720         94785   
Agriculture & Natural Resources                 34491         19860   
Industrial Arts & Consumer Services            104620         57200   
Communications & Journalism                    172992         86556   
Business                                       496570        148538   
Law & Public Policy                            100024         29221   

                                     Non_percentage  
Major_category                                       
Education                                  0.264425  
Engineering                                0.275112  
Health                                     0.310437  
Computers & Mathematics                    0.350708  
Physical Sciences                          0.392546  
Interdisciplinary                          0.429893  
Biology & Life Science                     0.456807  
Psychology & Social Work                   0.507584  
Humanities & Liberal Arts                  0.589392  
Social Science                             0.625385  
Arts                                       0.633334  
Agriculture & Natural Resources            0.634597  
Industrial Arts & Consumer Services        0.646521  
Communications & Journalism                0.666513  
Business                                   0.769747  
Law & Public Policy                        0.773910